--- a/apps/studio/pages/_app.tsx
+++ b/apps/studio/pages/_app.tsx
@@ -156,7 +156,10 @@ export default function App({ Component, pageProps }: AppProps) {
                     {/* Speed up initial API loading times by pre-connecting to the API domain */}
                     {IS_PLATFORM && (
                       <link
                         rel="preconnect"
-                        href={new URL(API_URL).origin}
+                        href={API_URL.startsWith('http')
+                          ? new URL(API_URL).origin
+                          : typeof window !== 'undefined'
+                            ? window.location.origin
+                            : ''}
                         crossOrigin="use-credentials"
                       />
                     )}
